### PR TITLE
Add JSONP support for monitoring routes

### PR DESCRIPTION
**Pull request by cmfatih**
See #100

### CODE REVIEW

**Comment by pkieltyka** [comment]
Also, the response content-type for jsonp is `application/javascript`

**Comment by cmfatih** [comment]
Damn finally...

File: server/monitor.go [hunk 1]
```diff
@@ -337,3 +341,19 @@ func updateUsage(v *Varz) {
 	v.CPU = pcpu
 	v.Cores = numCores
 }
+
+// ResponseHandler handles responses for monitoring routes
+func ResponseHandler(w http.ResponseWriter, r *http.Request, data []byte) {
+	// Get callback from request
+	callback := r.FormValue("callback")
```
Comment (pkieltyka) [inline server/monitor.go]: probably better off just using `r.URL.Query().Get("callback")` as you only need to check the URL query params

**Comment by cmfatih** [comment]
@pkieltyka : done, thank you :smile:

**Comment by derekcollison** [comment]
If not done already could someone update README to reflect the JSONP functionality?

**Comment by cmfatih** [comment]
@derekcollison : What kind of information you want to see in README related to this PR?

P.S.: Personally I'm thinking that `README.md` should be very lightweight for first impression. JSONP is a low level detail which should be in another file. Perhaps in a dedicated monitoring documentation or a file named `API.md` in root folder which contains everything about `gnatsd` with all the details.

**Comment by derekcollison** [comment]
The README has a monitoring section, just add a one liner that says it
supports JSONP via the callback arg and a simple example. No need for each
endpoint, just one IMO..

**Comment by derekcollison** [comment]
This good to go?

**Comment by cmfatih** [comment]
I just added a note to README about JSONP. I think it is okay to go...

**Pull request merged**